backend/app2/api/routes/auth.py

```
@@ -0,0 +1,113 @@
+            "access_token": result["access_token"],
+            "token_type": "bearer",
+            "user_id": result.get("user_id")
+        }
+    except Exception as e:
+        logger.error(f"Signup failed: {str(e)}")
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail=f"Registration failed: {str(e)}"
+        )
+
+@router.post("/login", response_model=TokenResponse)
```

Ruff (0.8.2)

Undefined name `TokenResponse`

(F821)